[Merge-Queue] Run `git svn fetch`

JonWBedard opened the pull request:
#### 81729c491c0a0702b84a9ea309eaf6ee8046d7d9
<pre>
[Merge-Queue] Run `git svn fetch`
<a href="https://bugs.webkit.org/show_bug.cgi?id=238759">https://bugs.webkit.org/show_bug.cgi?id=238759</a>
&lt;rdar://problem/91258277 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/factories.py:
(MergeQueueFactory.__init__): Add GitSvnFetch step.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Ditto.
* Tools/CISupport/ews-build/steps.py:
(UpdatePullRequest.hideStepIf):
(GitSvnFetch):
(GitSvnFetch.__init__): `git svn fetch` will have a non-zero exit code if the Subversion
record of changes does not match the git one, which will happen imiediately after committing.
(GitSvnFetch.getResultSummary):
(GitSvnFetch.evaluateCommand): Support expected failure.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/249352@main">https://commits.webkit.org/249352@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292509">https://svn.webkit.org/repository/webkit/trunk@292509</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4966,3 +4966,32 @@ def doStepIf(self, step):
+
+class GitSvnFetch(shell.ShellCommand):
+    name = 'git-svn-fetch'
+    haltOnFailure = False
```
should we also have flunkOnFailure = False

aj062 commented on Tools/CISupport/ews-build/factories.py:
```diff
@@ -326,6 +326,7 @@ def __init__(self, platform, configuration=None, architectures=None, additionalA
         self.addStep(PrintConfiguration())
         self.addStep(CleanGitRepo())
         self.addStep(CheckOutSource())
+        self.addStep(GitSvnFetch())
```
This is temporary until we move to git, right? Would be good to add a TODO to remove it once we fully move to git.

JonWBedard replied:
Yes, will add a to-do

JonWBedard committed:
[Merge-Queue] Run `git svn fetch`
https://bugs.webkit.org/show_bug.cgi?id=238759
<rdar://problem/91258277>

Patch by Jonathan Bedard <jbedard@apple.com> on 2022-04-06
Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/factories.py:
(MergeQueueFactory.__init__): Add GitSvnFetch step.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Ditto.
* Tools/CISupport/ews-build/steps.py:
(UpdatePullRequest.hideStepIf):
(GitSvnFetch):
(GitSvnFetch.__init__): `git svn fetch` will have a non-zero exit code if the Subversion
record of changes does not match the git one, which will happen imiediately after committing.
(GitSvnFetch.getResultSummary):
(GitSvnFetch.evaluateCommand): Support expected failure.
* Tools/CISupport/ews-build/steps_unittest.py:
